runtime/testy/stateManager.py

```diff
@@ -32,12 +32,52 @@ def addPipe(self, processName, pipe):
     self.processMapping[processName] = pipe
     pipe.send(RUNTIME_CONFIG.PIPE_READY.value)
 
-  def getValue(self, key):
-    self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(self.state[key])
+  def getValue(self, keys):
+    result = self.state
+    i = 0
+    try:
+      while (i < len(keys)):
+        result = result[keys[i]]
+        i += 1
+      self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(result)
+    except:
+      self.badThingsQueue.put(BadThing(sys.exc_info(), self.dictErrorMessage(i, keys, result), event = BAD_EVENTS.BAD_EVENT, printStackTrace = False))
+
+  def setValue(self, keys, value):
+    currDict = self.state
+    i = 0
+    try:
+      while (i < len(keys) - 1):
+        currDict = currDict[keys[i]]
+        i += 1
+      currDict[keys[i]] = value
+      self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(value)
+    except:
+      self.badThingsQueue.put(BadThing(sys.exc_info(), self.dictErrorMessage(i, keys, currDict), event = BAD_EVENTS.BAD_EVENT, printStackTrace = False))
+
+  def dictErrorMessage(self, erroredIndex, keys, currDict):
```

Comment:
The fact that you wrote this function for proper error messages is amazing. I love it.